src/test/kotlin/nl/info/zac/app/zaak/ZaakKoppelenRestServiceTest.kt

```diff
@@ -0,0 +1,117 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Lifely, 2025 Dimpact
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+package nl.info.zac.app.zaak
+
+import io.kotest.core.spec.style.BehaviorSpec
+import io.kotest.matchers.shouldBe
+import io.mockk.checkUnnecessaryStub
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import net.atos.client.zgw.shared.model.Archiefnominatie
+import net.atos.client.zgw.zrc.ZrcClientService
+import net.atos.zac.policy.PolicyService
+import nl.info.client.zgw.model.createZaak
+import nl.info.client.zgw.ztc.ZtcClientService
+import nl.info.zac.app.zaak.model.RelatieType
+import nl.info.zac.search.SearchService
+import nl.info.zac.search.model.ZoekResultaat
+import nl.info.zac.search.model.zoekobject.ZaakZoekObject
+import java.net.URI
+import java.util.UUID
+
+private const val OMSCHRIJVING = "fakeOmschrijving"
+
+private const val ZAAK_TYPE_OMSCHRIJVING = "Melding evenement organiseren behandelen"
+
+private const val STATUS_TYPE_OMSCHRIJVING = "Afgerond"
+
+@Suppress("LargeClass")
+class ZaakKoppelenRestServiceTest : BehaviorSpec({
+    val policyService = mockk<PolicyService>()
+    val searchService = mockk<SearchService>()
+    val zrcClientService = mockk<ZrcClientService>()
+    val ztcClientService = mockk<ZtcClientService>()
+    val zaakKoppelenRestService = ZaakKoppelenRestService(
+        policyService = policyService,
+        searchService = searchService,
+        zrcClientService = zrcClientService,
+        ztcClientService = ztcClientService
+    )
+
+    beforeEach {
+        checkUnnecessaryStub()
+    }
+
+    Given("A valid zaak UUID and relationType is provided") {
+        val zaakUuid = UUID.randomUUID()
+        val zaakZoekUuid = UUID.randomUUID().toString()
+        val relationType = RelatieType.HOOFDZAAK
+        val zoekZaakIdentifier = "ZAAK-2000-00002"
+        val zaakTypeURI = URI(UUID.randomUUID().toString())
+        val zaakZoekObjectTypeUuid1 = UUID.randomUUID().toString()
+        val page = 0
+        val rows = 10
+        val sourceZaak = createZaak(
+            uuid = zaakUuid,
+            identificatie = "ZAAK-2000-00001",
+            archiefnominatie = Archiefnominatie.BLIJVEND_BEWAREN,
+            zaakTypeURI = zaakTypeURI
+        )
+
+        val zaakZoekObject1 = ZaakZoekObject(
```

Comment:
We normally have `create` functions for this in our fixture files so that we have more consistency in creating such instances with less complexity in the unit tests. E.g. `createZaakZoekObject(..)` We already have that function so better to use that.